### PR TITLE
Nahiyan_Fix White Screen Dashboard Error

### PR DESCRIPTION
**Pull request by Nahiyan-16**
# Description
For the backend, I just made sure that we check is our cached data is up to date before returning it.
Before we didn't do this step and it resulted in added Project not being displayed in User Profile page.

## Related PRS (if any):
To test this backend PR you need to checkout the #XXX frontend PR.

## Main changes explained:
- Created an if statement to check is cached data is up to date.

## How to test:
1. check into current branch
2. do `npm install` ,`npm run build`, and `npm start` to run this PR locally
3. Clear site data/cache
4. log as Owner user
5. go to dashboard→ Other Links → Projects → Create a new project
6. Add your own account
7.  Add a WBS
8.  Add a task and assign to self
9.  Go to your User Profile -> Projects
10. Observe that your added project is being displayed

### CODE REVIEW

**Review by iven-yao — APPROVED** [review]
I just reviewed this pull request with its corresponding frontend, please check [here](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2091#pullrequestreview-1969530933) for more detail.

**Review by mohabbasd — APPROVED** [review]
Hello, I tested your PR and left a review on the frontend.

File: src/controllers/userProfileController.js [hunk 1]
```diff
@@ -594,10 +594,12 @@ const userProfileController = function (UserProfile) {
   const getUserById = function (req, res) {
     const userid = req.params.userId;
     if (cache.getCache(`user-${userid}`)) {
-      const getData = JSON.parse(cache.getCache(`user-${userid}`));
-      res.status(200).send(getData);
-      return;
-    }
+      const cachedData = JSON.parse(cache.getCache(`user-${userid}`));
+      if (cachedData.isUpToDate) {
```
Comment (abdel-lall) [inline src/controllers/userProfileController.js:598]: Hi, The property 'isUpToDate' does not exist in the user cached data. Consequently, this 'if' statement will always evaluate to false, which means 'getUserById' is no longer utilizing caching.

**Review by abdel-lall — CHANGES_REQUESTED** [review]

**Review by Sarthak32 — APPROVED** [review]
I have left the review on frontend PR : https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2091#pullrequestreview-1969880460

**Pull request closed without merging**